qt/main.cpp

```diff
@@ -165,6 +165,12 @@ int main(int argc, char * argv[])
 
 #if defined(OMIM_OS_MAC)
     apiOpenGLES3 = app.arguments().contains("es3", Qt::CaseInsensitive);
+#elif defined(OMIM_OS_LINUX)
+    // TODO: Implement proper runtime version detection in a separate commit
+    // Currently on Linux in a maximum ES2 scenario,
+    // the GL function pointers wouldn't be properly resolved anyway,
+    // so here at least a possibly successful path is chosen.
+    apiOpenGLES3 = true;
```

Comment:
Could you clarify here (I read flathub issue by diagonal), because according to the comment, current master should not work on Linux at all ..

We already have a bunch of Linux (Ubuntu) users that confirm that current master branch and desktop target works ok for them. Here we make completely new "branching" (before always ES2, now always ES3).

Reply:
The `master` works on linux as long as one has desktop GL available. The [linked issue](https://github.com/flathub/app.organicmaps.desktop/issues/1) is filled with reports from truly ES2 only devices where the code just doesn't work because (as it was also pointed out by biodranik) we are calling ES2 implementations of functions with parameters that are only supported on desktop GL but not on ES2. To be able to deal with those parameters we would need to resolve the function pointers to some `OES` and `EXT` extensions like the rest of the platforms, 
but the Linux code on the current master disregards the ES2 / ES3 flag and unconditionally [sets the function pointers without the needed extensions](https://github.com/organicmaps/organicmaps/blob/master/drape/gl_functions.cpp#L270) that [other platforms rely on in an ES2 scenario](https://github.com/organicmaps/organicmaps/blob/master/drape/gl_functions.cpp#L278).
Furthermore the ES2 headers are not even included, the `-lGLESv2` not even linked, and we don't have EGL, or any other dynamic resolution technique at place. So currently we don't even have the means to get the needed extensions for ES2. Luckily, so far, the tested ES3 implementations were able to take the passed parameters just like on every other platform.

Reply:
CHECK(false) in ES2 Linux.

Reply:
I have put some effort into dynamically resolving ES2 Extensions via EGL like the Android code does it, since most of the ES2 HW that run Linux are designed for Android anyway, and I guess it's OK to ignore Apple HW specifics for now.
So the reasons why I am not including it in this PR, is first of all to keep the PR size low, and easier to review.
And to give desktop wayland users [here](https://github.com/flathub/app.organicmaps.desktop/issues/1#issuecomment-1141671515) and [there](https://github.com/flathub/app.organicmaps.desktop/issues/1#issuecomment-1108203207) a quicker fix, while it is debated if EGL even is the right way forward or not. Plus I hit a bunch of weird linker issues ardound unity and the python bindings, which requires more investigation, so I am not even 100% sure if that path is even doable.